test/processPicker.test.ts

```diff
@@ -0,0 +1,193 @@
+        let pipeTransport = GetWindowsDockerLaunchJSONWithStringArgsAndDebuggerCommand();
+
+        // quoteArgs flag should be ignored
+        let pipeCmd = RemoteAttachPicker.createPipeCmdFromString(pipeTransport.pipeProgram, pipeTransport.pipeArgs, true);
+
+        pipeCmd.should.deep.equal("C:\\System32\\bash.exe -c docker -i exec 1234567 " + RemoteAttachPicker.scriptShellCmd);
+    });
+
+    test("Windows Docker with array args", () => {
+        let pipeTransport = GetWindowsDockerLaunchJSONWithArrayArgs();
+
+        let pipeCmd = RemoteAttachPicker.createPipeCmdFromArray(pipeTransport.pipeProgram, pipeTransport.pipeArgs, false);
+
+        pipeCmd.should.deep.equal("C:\\System32\\bash.exe -c docker -i exec 1234567 " + RemoteAttachPicker.scriptShellCmd);
+
+    });
+
+    test("OS Specific Configurations", () => {
+        let launch = GetOSSpecificJSON();
+
+        let pipeTransport = RemoteAttachPicker.getPipeTransportOptions(launch, "win32");
+
+        pipeTransport.pipeProgram.should.deep.equal("Windows pipeProgram");
+        pipeTransport.pipeArgs.should.deep.equal("windows");
+
+        pipeTransport = RemoteAttachPicker.getPipeTransportOptions(launch, "darwin");
+
+        pipeTransport.pipeProgram.should.deep.equal("OSX pipeProgram");
+        pipeTransport.pipeArgs.should.deep.equal(["osx"]);
+
+        pipeTransport = RemoteAttachPicker.getPipeTransportOptions(launch, "linux");
+
+        pipeTransport.pipeProgram.should.deep.equal("Linux pipeProgram");
+        // Linux pipeTransport does not have args defined, should use the one defined in pipeTransport.
+        pipeTransport.pipeArgs.should.deep.equal([]);
+
+    });
+});
+
+function GetWindowsWSLLaunchJSONWithArrayArgs() {
+    return {
+        pipeCwd: "${workspaceRoot}",
+        pipeProgram: "C:\\System32\\bash.exe",
+        pipeArgs: ["-c"]
+    }
+}
+
+function GetWindowsWSLLaunchJSONWithArrayArgsAndDebuggerCommand() {
+    return {
+        pipeCwd: "${workspaceRoot}",
+        pipeProgram: "C:\\System32\\bash.exe",
+        pipeArgs: ["-c", "${debuggerCommand}", "--", "ignored"]
+    }
+}
+
+function GetWindowsWSLLaunchJSONWithStringArgs() {
+    return {
+        pipeCwd: "${workspaceRoot}",
+        pipeProgram: "C:\\System32\\bash.exe",
+        pipeArgs: "-c"
+    }
+}
+
+function GetWindowsWSLLaunchJSONWithStringArgsAndDebuggerCommand() {
+    return {
+        pipeCwd: "${workspaceRoot}",
+        pipeProgram: "C:\\System32\\bash.exe",
+        pipeArgs: "-c ${debuggerCommand} -- ignored"
+    }
+}
+
+function GetWindowsDockerLaunchJSONWithArrayArgs() {
+    return {
+        pipeCwd: "${workspaceRoot}",
+        pipeProgram: "C:\\System32\\bash.exe",
+        pipeArgs: ["-c", "docker", "-i", "exec", "1234567"]
+    }
+};
+
+function GetWindowsDockerLaunchJSONWithStringArgsAndDebuggerCommand() {
+    return {
+        pipeCwd: "${workspaceRoot}",
+        pipeProgram: "C:\\System32\\bash.exe",
```

Comment:
>pipeProgram: "C:\\System32\\bash.exe", [](start = 8, length = 38)

Not sure if it really matters, but this isn't the right way to do docker launch anymore. The pipeProgram should be 'docker' and `"quoteArgs": false` should be used.